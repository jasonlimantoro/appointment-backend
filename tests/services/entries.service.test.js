import AWS from 'aws-sdk';
import EntryService from '../../services/entry.service';
import mockedData from '../../fixtures/entries';

const localDynamo = new AWS.DynamoDB.DocumentClient({
  region: 'localhost',
  endpoint: process.env.dynamoDBEndPoint,
});
const service = new EntryService({ dataSource: localDynamo });

describe('Entry Service', () => {
  it('list: should work', async () => {
    const res = await service.list();
    expect(res.length).toEqual(3);
    expect(res).toMatchSnapshot();
  });
});
