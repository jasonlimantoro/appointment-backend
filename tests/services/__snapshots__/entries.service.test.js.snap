// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Service list: should work 1`] = `
Array [
  Object {
    "Guest": Object {
      "NIK": "12312365",
      "company": "IMCP",
      "email": "andy@gmail.com",
      "firstName": "Andy",
      "id": "ah",
      "lastName": "Huang",
    },
    "id": "234",
    "see": "person2",
  },
  Object {
    "Guest": Object {
      "NIK": "12312365",
      "company": "IMCP",
      "email": "andy@gmail.com",
      "firstName": "Andy",
      "id": "ah",
      "lastName": "Huang",
    },
    "id": "123",
    "see": "person1",
  },
  Object {
    "Guest": Object {
      "NIK": "12832837",
      "company": "Google",
      "email": "budy@gmail.com",
      "firstName": "Budy",
      "id": "bh",
      "lastName": "Harjo",
    },
    "id": "456k",
    "see": "person2",
  },
]
`;
